### DISCUSSION TITLE
How to find the user details with the default `GITHUB_TOKEN` in GitHub Actions?

### DISCUSSION THREAD

**Discussion by retzero** [actions/runner#3289] [Q&A]
Hello.

There exist default `GITHUB_TOKEN` secrets which can be used during the workflows run.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

How can I retrieve user information from this token?

I've tried this api but it is not working as expected...

```
curl -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/user
```

```
{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/actions#get-github-actions-permissions-for-an-organization"
}
```

**Comment by enescakir** [comment]
What do you mean by 'user information'? If you're looking to identify who initiated the workflow, you can use [the GitHub context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) or [environment variable](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) in the workflow.

```shell
    echo "$GITHUB_ACTOR"
    echo "${{ github.actor }}"
```

**Comment by enescakir** [comment]
You can also obtain `workflow_job` details, which include a variety of data, by sending a request to [Get a workflow run](https://docs.github.com/en/rest#get-a-workflow-run)

```bash
curl -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/repos/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID

```
